File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        if(arr.length == 1) return new int[]{-1};
        
        int[] answer = new int[arr.length - 1];
        int minValue = Integer.MAX_VALUE;
        
        for(int num : arr) {
            if(minValue > num) minValue = num;
        }
        
        int idx =0;
        for(int num: arr) {
            if(num != minValue) answer[idx++] = num;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.go
func solution(arr []int) []int {
    if len(arr) == 1 {
        return []int{-1}
    }
    
    minIdx := 0
    minValue := arr[minIdx]
    
    for i,v := range arr {
        if minValue > v {
            minIdx = i
            minValue = v
        }
    }
    
    
    return append(arr[:minIdx], arr[minIdx+1:]...)
}